runtime/doc/lua.txt

```diff
@@ -1996,6 +1996,20 @@ vim.list_slice({list}, {start}, {finish})                   *vim.list_slice()*
     Return: ~
         (list) Copy of table sliced from start to finish (inclusive)
 
+vim.memoize({fn}, {hash_fn})                                   *vim.memoize()*
+    Memoize a function {fn} using {hash_fn} to hash the arguments. Internally
+    uses a weak table to cache the results of {fn}.
+
+    Parameters: ~
+      • {fn}       (function) Function to memoize.
+      • {hash_fn}  integer|'concat'|function|nil Hash function to create a
+                   hash to use as a key to store results. When an integer,
+                   refers to the argument index to use as the hash. When 'concat `, the hash is determined by string concatenating all the
```

Comment:
```suggestion
                   refers to the argument index to use as the hash. When `'concat'`, the hash is determined by string concatenating all the
```

Hmmm is the type generator script messing up the quotes?

Reply:
Typo: mixed quotes?

Reply:
No, it's just bad docgen.

Reply:
Ah, I see now, the quotes in the type are unexpected. Should be `string`, anyway, as that's the type, and the text afterwards should make clear that `'concat'` is the only valid string?

Reply:
The type is ok. Docgen doesn't like `\`'concat'\``.